orchestrator/modules/tools/tool_router.py

```
@@ -939,6 +952,31 @@ async def get_filtered_composio_actions(
+    try:
+        from config import config
+
+        if not bool(getattr(config, "COMPOSIO_DESTRUCTIVE_FAIL_CLOSED", True)):
+            return False
+    except Exception:
+        pass  # default to fail-closed behavior if config can't be read
+    try:
+        from modules.tools.capabilities.taxonomy import intent_is_destructive
+
+        return intent_is_destructive(intent)
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)